src/coreclr/jit/targetloongarch64.cpp

```diff
@@ -24,4 +24,221 @@ const regNumber fltArgRegs [] = {REG_F0, REG_F1, REG_F2, REG_F3, REG_F4, REG_F5,
+                    canPassArgInRegisters = m_floatRegs.Count() > 0;
+
+                    argRegTypeInStruct1 = (passedSize == 8) ? TYP_DOUBLE : TYP_FLOAT;
+                }
+                else if ((floatFlags & STRUCT_FLOAT_FIELD_ONLY_TWO) != 0)
+                {
+                    slots                 = 2;
+                    canPassArgInRegisters = m_floatRegs.Count() >= 2;
+
+                    argRegTypeInStruct1 = (floatFlags & STRUCT_FIRST_FIELD_SIZE_IS8) ? TYP_DOUBLE : TYP_FLOAT;
+                    argRegTypeInStruct2 = (floatFlags & STRUCT_SECOND_FIELD_SIZE_IS8) ? TYP_DOUBLE : TYP_FLOAT;
+                }
+                else if ((floatFlags & STRUCT_FLOAT_FIELD_FIRST) != 0)
+                {
+                    slots                 = 1;
+                    canPassArgInRegisters = (m_floatRegs.Count() > 0) && (m_intRegs.Count() > 0);
+
+                    argRegTypeInStruct1 = (floatFlags & STRUCT_FIRST_FIELD_SIZE_IS8) ? TYP_DOUBLE : TYP_FLOAT;
+                    argRegTypeInStruct2 = (floatFlags & STRUCT_SECOND_FIELD_SIZE_IS8) ? TYP_LONG : TYP_INT;
+                }
+                else if ((floatFlags & STRUCT_FLOAT_FIELD_SECOND) != 0)
+                {
+                    slots                 = 1;
+                    canPassArgInRegisters = (m_floatRegs.Count() > 0) && (m_intRegs.Count() > 0);
+
+                    argRegTypeInStruct1 = (floatFlags & STRUCT_FIRST_FIELD_SIZE_IS8) ? TYP_LONG : TYP_INT;
+                    argRegTypeInStruct2 = (floatFlags & STRUCT_SECOND_FIELD_SIZE_IS8) ? TYP_DOUBLE : TYP_FLOAT;
+                }
+
+                assert((slots == 1) || (slots == 2));
+
+                if (!canPassArgInRegisters)
+                {
+                    m_floatRegs.Clear();
+                    slots = (passedSize + TARGET_POINTER_SIZE - 1) / TARGET_POINTER_SIZE;
+                    // On LoongArch64, if there aren't any remaining floating-point registers to pass the argument,
+                    // integer registers (if any) are used instead.
+                    canPassArgInRegisters = m_intRegs.Count() >= slots;
+
+                    argRegTypeInStruct1 = TYP_UNKNOWN;
+                    argRegTypeInStruct2 = TYP_UNKNOWN;
+                }
+            }
+            else
+            {
+                slots                 = (passedSize + TARGET_POINTER_SIZE - 1) / TARGET_POINTER_SIZE;
+                canPassArgInRegisters = m_intRegs.Count() >= slots;
+            }
+
+            if (!canPassArgInRegisters && (slots == 2))
+            {
+                // Here a struct-arg which needs two registers but only one integer register available,
+                // it has to be split.
+                if (m_intRegs.Count() > 0)
+                {
+                    canPassArgInRegisters = true;
+                }
+            }
+        }
+        else
+        {
+            // how about the IsBlockLayout?  liking the class type ? just as pointer.
```

Comment:
You can add `assert(!structLayout->IsBlockLayout())`. ABI classifiers will never see those kind of layouts, we always have a class handle from the signature to classify with.

Reply:
(similar for the RISC-V classifier, cc @tomeksowi)

Reply:
OK, thanks